### PR TITLE
TW-403: add share screen and forward screen like design

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo mobile:
https://github.com/linagora/twake-on-matrix/assets/43041967/b9db3a9f-fba2-4ed8-a4dc-b92063dd912d

https://github.com/linagora/twake-on-matrix/assets/43041967/2cfa1904-1e68-47f5-9a80-4344c10c99c6

### Demo web:

https://github.com/linagora/twake-on-matrix/assets/43041967/b0946f00-b74d-489e-bee0-49224be6421a

### CODE REVIEW

File: lib/pages/forward/forward_view_style.dart [hunk 1]
```diff
@@ -3,9 +3,9 @@ import 'package:flutter/cupertino.dart';
...
   static double preferredAppBarSize(BuildContext context) => 64;
 
-  static double get paddingBody => 16.0;
+  static double get paddingBody => 8.0;
 
-  static double get bottomBarHeight => 120.0;
+  static double get bottomBarHeight => 60.0;
 
-  static double get iconSendSize => 40.0;
+  static double get iconSendSize => 56.0;
```
Comment (drminh2807) [inline lib/pages/forward/forward_view_style.dart:10]:
```suggestion
  static const int iconSendSize = 56.0;
```
SHould use const to increase build performance

File: lib/pages/share/share.dart [hunk 2]
```diff
@@ -42,6 +48,8 @@ class ShareController extends State<Share> with SendFilesMixin {
     isShowRecentlyChatsNotifier.value = !isShowRecentlyChatsNotifier.value;
   }
 
+  void toggleSearchMode() {}
```
Comment (drminh2807) [inline lib/pages/share/share.dart:51]: empty ?

File: lib/pages/share/share_view.dart [hunk 3]
```diff
@@ -16,28 +18,22 @@ class ShareView extends StatelessWidget {
...
+          title: L10n.of(context)!.selectChat,
+          searchModeNotifier: controller.isSearchModeNotifier,
+          textEditingController: controller.textEditingController,
+          toggleSearchMode: controller.toggleSearchMode,
+          focusNode: controller.searchFocusNode,
         ),
       ),
       body: Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 16.0),
+        padding: const EdgeInsets.symmetric(horizontal: 8.0, vertical: 8.0),
```
Comment (drminh2807) [inline lib/pages/share/share_view.dart:32]:
```suggestion
        padding: const EdgeInsets.all(8.0),
```

File: lib/pages/chat_list/chat_list_item.dart [hunk 4]
```diff
@@ -131,332 +119,252 @@ class ChatListItem extends StatelessWidget {
...
-                    overflow: TextOverflow.ellipsis,
-                    maxLines: 1,
-                    softWrap: false,
-                    style: Theme.of(context).textTheme.labelLarge?.merge(
-                          TextStyle(
-                            overflow: TextOverflow.ellipsis,
-                            color: unread
-                                ? Theme.of(context).colorScheme.onSurface
-                                : ChatListItemStyle.readMessageColor,
+class ChatListItemSubtitle extends StatelessWidget with ChatListItemMixin {
```
[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat_list/chat_list_item.dart:212]: i use it in another file

File: lib/pages/share/share.dart (hunk 2 above)
Comment (sherlockvn) [reply lib/pages/share/share.dart:51]: yeah, current not support search, but i reuse the old component, so the method need to have to be able to reused

File: lib/pages/forward/recent_chat_list.dart [hunk 5]
```diff
@@ -30,13 +33,52 @@ class RecentChatList extends StatelessWidget {
...
+                    Avatar(
+                      mxContent: room.avatar,
+                      name: room.getLocalizedDisplayname(
+                        MatrixLocals(L10n.of(context)!),
+                      ),
+                      onTap: null,
+                    ),
+                    Expanded(
+                      child: Padding(
+                        padding: const EdgeInsets.only(
```
Comment (nqhhdev) [inline lib/pages/forward/recent_chat_list.dart:65]: Pls use `EdgeInsetsDirectional`

File: lib/pages/forward/forward_view.dart [hunk 6]
```diff
@@ -38,14 +38,10 @@ class ForwardView extends StatelessWidget {
           return true;
         },
         child: SingleChildScrollView(
-          padding: EdgeInsets.all(ForwardViewStyle.paddingBody),
+          padding: const EdgeInsets.all(ForwardViewStyle.paddingBody),
```
Comment (nqhhdev) [inline lib/pages/forward/forward_view.dart:41]: Use `EdgeInsetsDirectional`

File: lib/pages/chat_list/chat_list_item.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (drminh2807) [reply lib/pages/chat_list/chat_list_item.dart:213]: okay, so you should create another file to contains only one widget class

[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat_list/chat_list_item.dart:213]: Done

File: lib/pages/chat_list/chat_list_item_subtitle.dart [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+import 'package:fluffychat/pages/chat_list/chat_list_item_mixin.dart';
+import 'package:fluffychat/pages/chat_list/chat_list_item_style.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_locals.dart';
+import 'package:fluffychat/utils/room_status_extension.dart';
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:matrix/matrix.dart';
+
+import '../../config/themes.dart';
```
Comment (drminh2807) [inline lib/pages/chat_list/chat_list_item_subtitle.dart:11]: pleas use absolute path

**Comment by nqhhdev** [comment]
1. I can't back to previous screen when click on back in Forward Screen (Web version).
2. Open forward screen => click on  a conversation in chat list => can't open a conversation

Pls check it again

https://github.com/linagora/twake-on-matrix/assets/99852347/b34e234c-0bfb-4eef-b793-914dfdf5a73d

**Review by drminh2807 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**